pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -1622,6 +1626,17 @@ func validateDisks(field *k8sfield.Path, disks []v1.Disk) []metav1.StatusCause {
 				Field:   field.Index(idx).Child("cache").String(),
 			})
 		}
+
+		// Verify disk and volume name can be a valid container name since disk
+		// name can become a container name which will fail to schedule if invalid
+		e := validation.IsDNS1123Subdomain(disk.Name)
+		if len(e) != 0 {
+			causes = append(causes, metav1.StatusCause{
```

Comment:
I think you need to iterate the returned errors, and add a cause for everyone (dns formatting and length).